core/sql/optimizer/BindRI.cpp

```diff
@@ -408,6 +408,39 @@ void RefConstraint::getMatchOptionPredicateText(NAString &text,
   text += ")";
 }
 
+//helper function to check if the given column name is reserved hidden coloum
+static NABoolean isHiddenColumn(const char *colname)
+{
+  int len = strlen(colname);
+  if(strcmp(colname , "_SALT_") ==0) 
+    return TRUE;
+  //check for DIVISION column
+  //pattern _DIVISION_%d_
+  //must longer than 12
+  if(len >= 12) {
+    //must end with _
+    if(colname[len] == '_')
```

Comment:
colname[len] will always be '\0' since len is the strlen of colname. I think you meant, colname[len-1] here.